File: frontend/src/types/system.ts
export interface LoginRequest {
  username: string;
  password: string;
}

export interface RoleSummary {
  id: number;
  name: string;
  code: string;
  description: string | null;
  is_system: boolean;
  menu_ids: number[];
}

export interface UserProfile {
  id: number;
  username: string;
  email: string | null;
  phone: string | null;
  nickname: string | null;
  status: "ACTIVE" | "DISABLED";
  is_superuser: boolean;
  roles: RoleSummary[];
  permissions: string[];
  last_login_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface LoginResponse {
  access_token: string;
  token_type: "bearer";
  expires_in: number;
  user: UserProfile;
}

export interface RoleItem {
  id: number;
  name: string;
  code: string;
  description: string | null;
  is_system: boolean;
  menu_ids: number[];
}

export interface MenuNode {
  id: number;
  parent_id: number | null;
  name: string;
  title: string;
  path: string | null;
  component: string | null;
  icon: string | null;
  permission_code: string | null;
  menu_type: "DIRECTORY" | "MENU" | "BUTTON";
  sort_order: number;
  hidden: boolean;
  created_at: string;
  updated_at: string;
  children: MenuNode[];
}

export interface SystemConfigItem {
  id: number;
  category: string;
  config_key: string;
  config_value: string;
  value_type: "STRING" | "INTEGER" | "FLOAT" | "BOOLEAN" | "JSON" | "SECRET";
  description: string | null;
  is_secret: boolean;
  is_enabled: boolean;
  created_at: string;
  updated_at: string;
}

export interface PlatformAccountItem {
  id: number;
  platform: string;
  display_name: string;
  account_id: string;
  auth_status: "UNAUTHORIZED" | "AUTHORIZED" | "EXPIRED" | "DISABLED";
  access_token: string | null;
  refresh_token: string | null;
  expires_at: string | null;
  extra: Record<string, unknown> | null;
  is_enabled: boolean;
  created_at: string;
  updated_at: string;
}

export interface OperationLogItem {
  id: number;
  module: string;
  action: string;
  biz_type: string | null;
  biz_id: string | null;
  operator_id: number | null;
  operator_name: string | null;
  request_id: string | null;
  method: string | null;
  path: string | null;
  ip_address: string | null;
  status: "SUCCESS" | "FAILED";
  message: string | null;
  details: Record<string, unknown> | null;
  created_at: string;
}

export interface ErrorLogItem {
  id: number;
  request_id: string | null;
  path: string | null;
  method: string | null;
  user_id: number | null;
  error_code: number;
  error_type: string;
  error_message: string;
  stack_trace: string | null;
  created_at: string;
}

export interface TaskJobItem {
  id: number;
  task_name: string;
  task_type: string;
  queue_name: string;
  status: "PENDING" | "RUNNING" | "SUCCESS" | "FAILED" | "RETRYING" | "CANCELLED";
  celery_task_id: string | null;
  request_id: string | null;
  triggered_by: number | null;
  payload: Record<string, unknown> | null;
  result: Record<string, unknown> | null;
  progress: number;
  error_message: string | null;
  retry_count: number;
  started_at: string | null;
  finished_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface FileAssetItem {
  id: number;
  original_name: string;
  storage_name: string;
  category: "IMAGE" | "AUDIO" | "VIDEO" | "DOCUMENT" | "OTHER";
  content_type: string;
  extension: string | null;
  size: number;
  storage_backend: string;
  storage_path: string;
  url: string;
  bucket_name: string | null;
  uploaded_by: number | null;
  created_at: string;
  updated_at: string;
}

export interface HealthInfo {
  service: string;
  environment: string;
  api_prefix: string;
}

export interface LoginPayload {
  username: string;
  password: string;
}

export interface UserUpsertForm {
  id?: number;
  username?: string;
  password?: string;
  email?: string | null;
  phone?: string | null;
  nickname?: string | null;
  role_ids: number[];
  is_superuser: boolean;
  status: "ACTIVE" | "DISABLED";
}

export interface RoleUpsertForm {
  id?: number;
  name: string;
  code: string;
  description?: string | null;
  menu_ids: number[];
  is_system: boolean;
}

export interface MenuUpsertForm {
  id?: number;
  parent_id: number | null;
  name: string;
  title: string;
  path?: string | null;
  component?: string | null;
  icon?: string | null;
  permission_code?: string | null;
  menu_type: "DIRECTORY" | "MENU" | "BUTTON";
  sort_order: number;
  hidden: boolean;
}

export interface ConfigUpsertForm {
  id?: number;
  category: string;
  config_key: string;
  config_value: string;
  value_type: "STRING" | "INTEGER" | "FLOAT" | "BOOLEAN" | "JSON" | "SECRET";
  description?: string | null;
  is_secret: boolean;
  is_enabled: boolean;
}

export interface PlatformAccountUpsertForm {
  id?: number;
  platform: string;
  display_name: string;
  account_id: string;
  auth_status: "UNAUTHORIZED" | "AUTHORIZED" | "EXPIRED" | "DISABLED";
  access_token?: string | null;
  refresh_token?: string | null;
  expires_at?: string | null;
  extra?: Record<string, unknown> | null;
  is_enabled: boolean;
}

export interface DemoTaskPayload {
  payload: {
    message: string;
  };
}

export interface DashboardOverview {
  health: HealthInfo;
  metrics: Array<{
    label: string;
    value: string;
    hint: string;
    trend?: string;
  }>;
  activity: OperationLogItem[];
  recent_tasks: TaskJobItem[];
}

File: backend/app/db/models/system.py
from datetime import UTC, datetime

from sqlalchemy import Boolean, DateTime, Index, Integer, JSON, String, Text, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from app.core.constants import ConfigValueType, OperationStatus, PlatformAuthStatus, TaskStatus
from app.db.models.base import Base
from app.db.models.mixins import IDMixin, ID_TYPE, TimestampMixin


class OperationLog(IDMixin, Base):
    __tablename__ = "operation_logs"
    __table_args__ = (
        Index("ix_operation_logs_module", "module"),
        Index("ix_operation_logs_operator_id", "operator_id"),
        Index("ix_operation_logs_created_at", "created_at"),
    )

    module: Mapped[str] = mapped_column(String(64), nullable=False)
    action: Mapped[str] = mapped_column(String(64), nullable=False)
    biz_type: Mapped[str | None] = mapped_column(String(64))
    biz_id: Mapped[str | None] = mapped_column(String(64))
    operator_id: Mapped[int | None] = mapped_column(ID_TYPE)
    operator_name: Mapped[str | None] = mapped_column(String(128))
    request_id: Mapped[str | None] = mapped_column(String(64))
    method: Mapped[str | None] = mapped_column(String(16))
    path: Mapped[str | None] = mapped_column(String(255))
    ip_address: Mapped[str | None] = mapped_column(String(64))
    status: Mapped[str] = mapped_column(String(32), default=OperationStatus.SUCCESS.value, nullable=False)
    message: Mapped[str | None] = mapped_column(String(255))
    details: Mapped[dict | None] = mapped_column(JSON)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        default=lambda: datetime.now(UTC),
    )


class ErrorLog(IDMixin, Base):
    __tablename__ = "error_logs"
    __table_args__ = (
        Index("ix_error_logs_request_id", "request_id"),
        Index("ix_error_logs_created_at", "created_at"),
    )

    request_id: Mapped[str | None] = mapped_column(String(64))
    path: Mapped[str | None] = mapped_column(String(255))
    method: Mapped[str | None] = mapped_column(String(16))
    user_id: Mapped[int | None] = mapped_column(ID_TYPE)
    error_code: Mapped[int] = mapped_column(Integer, nullable=False)
    error_type: Mapped[str] = mapped_column(String(128), nullable=False)
    error_message: Mapped[str] = mapped_column(Text, nullable=False)
    stack_trace: Mapped[str | None] = mapped_column(Text)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        default=lambda: datetime.now(UTC),
    )


class TaskJob(IDMixin, TimestampMixin, Base):
    __tablename__ = "task_jobs"
    __table_args__ = (
        Index("ix_task_jobs_status", "status"),
        Index("ix_task_jobs_task_name", "task_name"),
        Index("ix_task_jobs_request_id", "request_id"),
    )

    task_name: Mapped[str] = mapped_column(String(128), nullable=False)
    task_type: Mapped[str] = mapped_column(String(64), nullable=False)
    queue_name: Mapped[str] = mapped_column(String(64), nullable=False)
    status: Mapped[str] = mapped_column(String(32), default=TaskStatus.PENDING.value, nullable=False)
    celery_task_id: Mapped[str | None] = mapped_column(String(128))
    request_id: Mapped[str | None] = mapped_column(String(64))
    triggered_by: Mapped[int | None] = mapped_column(ID_TYPE)
    payload: Mapped[dict | None] = mapped_column(JSON)
    result: Mapped[dict | None] = mapped_column(JSON)
    progress: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    error_message: Mapped[str | None] = mapped_column(Text)
    retry_count: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    started_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    finished_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))


class FileAsset(IDMixin, TimestampMixin, Base):
    __tablename__ = "file_assets"
    __table_args__ = (
        Index("ix_file_assets_category", "category"),
        Index("ix_file_assets_uploaded_by", "uploaded_by"),
    )

    original_name: Mapped[str] = mapped_column(String(255), nullable=False)
    storage_name: Mapped[str] = mapped_column(String(255), nullable=False)
    category: Mapped[str] = mapped_column(String(32), nullable=False)
    content_type: Mapped[str] = mapped_column(String(128), nullable=False)
    extension: Mapped[str | None] = mapped_column(String(32))
    size: Mapped[int] = mapped_column(Integer, nullable=False)
    storage_backend: Mapped[str] = mapped_column(String(64), nullable=False)
    storage_path: Mapped[str] = mapped_column(String(500), nullable=False)
    url: Mapped[str] = mapped_column(String(500), nullable=False)
    bucket_name: Mapped[str | None] = mapped_column(String(128))
    uploaded_by: Mapped[int | None] = mapped_column(ID_TYPE)


class PlatformAccount(IDMixin, TimestampMixin, Base):
    __tablename__ = "platform_accounts"
    __table_args__ = (
        UniqueConstraint("platform", "account_id", name="uq_platform_accounts_platform_account"),
        Index("ix_platform_accounts_platform", "platform"),
    )

    platform: Mapped[str] = mapped_column(String(64), nullable=False)
    display_name: Mapped[str] = mapped_column(String(128), nullable=False)
    account_id: Mapped[str] = mapped_column(String(128), nullable=False)
    auth_status: Mapped[str] = mapped_column(
        String(32),
        default=PlatformAuthStatus.UNAUTHORIZED.value,
        nullable=False,
    )
    access_token: Mapped[str | None] = mapped_column(Text)
    refresh_token: Mapped[str | None] = mapped_column(Text)
    expires_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    extra: Mapped[dict | None] = mapped_column(JSON)
    is_enabled: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)


class SystemConfig(IDMixin, TimestampMixin, Base):
    __tablename__ = "system_configs"
    __table_args__ = (
        UniqueConstraint("category", "config_key", name="uq_system_configs_category_key"),
        Index("ix_system_configs_category", "category"),
    )

    category: Mapped[str] = mapped_column(String(64), nullable=False)
    config_key: Mapped[str] = mapped_column(String(128), nullable=False)
    config_value: Mapped[str] = mapped_column(Text, nullable=False)
    value_type: Mapped[str] = mapped_column(
        String(32),
        default=ConfigValueType.STRING.value,
        nullable=False,
    )
    description: Mapped[str | None] = mapped_column(String(255))
    is_secret: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    is_enabled: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
